includes/CreateWikiRegexConstraint.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace Miraheze\CreateWiki;
+
+use StringUtils;
+
+class CreateWikiRegexConstraint {
+
+	/**
+	 * @param array $regexes
+	 * @return bool
+	 */
+	private static function validateRegexes( $regexes ) {
+		foreach ( $regexes as $regex ) {
+			if ( !StringUtils::isValidPCRERegex( $regex ) ) {
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * @param array &$regexes
+	 */
+	private static function filterInvalidRegexes( &$regexes ) {
+		foreach ( $regexes as $regex ) {
+			if ( !StringUtils::isValidPCRERegex( $regex ) ) {
+				wfWarn( 'Contains invalid regex.' );
+
+				unset( $regexes[$regex] );
+			}
+		}
```

Comment:
I belive the key of the element should be used to unset it:
```suggestion
		foreach ( $regexes as $key => $regex ) {
			if ( !StringUtils::isValidPCRERegex( $regex ) ) {
				wfWarn( 'Contains invalid regex.' );

				unset( $regexes[$key] );
			}
		}
```

Reply:
Done. Thanks!